src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] values;
+    private int size;
+
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private void resize() {
+        if (size == values.length) {
+            T[] buffer = (T[]) new Object[values.length + values.length / 2];
+            System.arraycopy(values, 0, buffer, 0, size);
+            values = buffer;
+        }
+    }
+
+    private void indexCheck(int index, int rangeMax) {
+        if (index < 0 || index > rangeMax) {
+            throw new ArrayListIndexOutOfBoundsException("The passed index is wrong");
```

Comment:
```suggestion
            throw new ArrayListIndexOutOfBoundsException("The passed index is wrong " + index);
```

Reply:
✅
The message was changed to ("The passed index " + index + " is wrong")